Upstream-Status: Submitted

This patch adds the fix for one of the ciphers used in openssl, namely
the cipher des-ede3-cfb1. Complete bug log and patch is present here:
http://rt.openssl.org/Ticket/Display.html?id=2867

Signed-Off-By: Muhammad Shakeel <muhammad_shakeel@mentor.com>

ported the patch to the 1.0.0m version
Signed-off-by: Brendan Le Foll <brendan.le.foll@intel.com> 2015/03/24

diff --git a/crypto/evp/e_des3.c b/crypto/evp/e_des3.c
index 3232cfe..df84922 100644
===================================================================
--- a/crypto/evp/e_des3.c
+++ b/crypto/evp/e_des3.c
@@ -181,7 +181,7 @@ static int des_ede3_cfb1_cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
     size_t n;
     unsigned char c[1], d[1];
 
-    for (n = 0; n < inl; ++n) {
+    for (n = 0; n < inl * 8; ++n) {
         c[0] = (in[n / 8] & (1 << (7 - n % 8))) ? 0x80 : 0;
         DES_ede3_cfb_encrypt(c, d, 1, 1,
                              &data(ctx)->ks1, &data(ctx)->ks2,
